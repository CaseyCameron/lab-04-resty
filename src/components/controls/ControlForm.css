.Form {
  margin-right: 0.3vw;
  width: 81.9vw;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;

}

.Form section {
  display: flex;
  width: 100%;
  justify-content: flex-start;
  align-items: center;
  flex-direction: row;
}

.Form section input[type="radio"] {
  display: flex;
  height: 0;
  width: 0;
  margin-left: 0;
}

.Form section label {
  padding: 4px 16px;
  cursor: pointer;
}

.Form section input[type="radio"]:checked + label {
  display: hidden;
  background-color: #204060;
  color: white;
}

.Form input {
  margin-bottom: 8px;
}
.Form button {
  margin-left: 2vw;
}
.Form textarea {
  margin-top: 5px;
  width: 50%;
  height: 6vw;
  align-self: flex-start;
}


