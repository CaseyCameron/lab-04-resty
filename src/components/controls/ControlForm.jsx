import React from 'react';
import PropTypes from 'prop-types';
import style from './ControlForm.css';

const ControlForm = ({
  urlValue,
  handleURLChange,
  handleMethod,
  method,
  bodyValue,
  handleBodyValueChange,
  handleSubmit
}) => {

  return (
    <section>
      <form className={style.Form} aria-label='api-form' onSubmit={handleSubmit}>
        <p>https://last-airbender-api.herokuapp.com/api/v1/characters/5cf5679a915ecad153ab68c8</p>
        <input
          style={{ width: '120vh' }}
          className='url'
          value={urlValue}
          type='text'
          placeholder='URL'
          onChange={handleURLChange}
        />
        <br />
        <section>
          <input type='radio' name='method' value='GET' checked={method === 'GET'} onChange={handleMethod} />
          <label>GET</label>
          <input type='radio' name='method' value='POST' checked={method === 'POST'} onChange={handleMethod} />
          <label>POST</label>
          <input type='radio' name='method' value='PUT' checked={method === 'PUT'} onChange={handleMethod} />
          <label>PUT</label>
          <input type='radio' name='method' value='PATCH' checked={method === 'PATCH'} onChange={handleMethod} />
          <label>PATCH</label>
          <input type='radio' name='method' value='DELETE' checked={method === 'DELETE'} onChange={handleMethod} />
          <label>DELETE</label>
          <button name='submit'>Go!</button>
        </section>
        <br />
        <textarea value={bodyValue} type='text' placeholder='Raw JSON Body' onChange={handleBodyValueChange} />
      </form>
    </section >
  );
};

ControlForm.propTypes = {
  urlValue: PropTypes.string.isRequired,
  handleURLChange: PropTypes.func.isRequired,
  handleMethod: PropTypes.func.isRequired,
  method: PropTypes.string.isRequired,
  bodyValue: PropTypes.string.isRequired,
  handleBodyValueChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

export default ControlForm;