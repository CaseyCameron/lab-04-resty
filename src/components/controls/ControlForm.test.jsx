import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ControlForm from '../../components/controls/ControlForm';

describe('Form component', () => {
  afterEach(() => cleanup());
  it('Renders the form', () => {
    const { asFragment } = render(<ControlForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});