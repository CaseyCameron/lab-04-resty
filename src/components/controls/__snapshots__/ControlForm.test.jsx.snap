// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component Renders the form 1`] = `
<DocumentFragment>
  <section>
    <form
      aria-label="api-form"
      class="Form"
    >
      <p>
        https://last-airbender-api.herokuapp.com/api/v1/characters/5cf5679a915ecad153ab68c8
      </p>
      <input
        class="url"
        placeholder="URL"
        style="width: 120vh;"
        type="text"
        value=""
      />
      <br />
      <section>
        <input
          name="method"
          type="radio"
          value="GET"
        />
        <label>
          GET
        </label>
        <input
          name="method"
          type="radio"
          value="POST"
        />
        <label>
          POST
        </label>
        <input
          name="method"
          type="radio"
          value="PUT"
        />
        <label>
          PUT
        </label>
        <input
          name="method"
          type="radio"
          value="PATCH"
        />
        <label>
          PATCH
        </label>
        <input
          name="method"
          type="radio"
          value="DELETE"
        />
        <label>
          DELETE
        </label>
        <button
          name="submit"
        >
          Go!
        </button>
      </section>
      <br />
      <textarea
        placeholder="Raw JSON Body"
        type="text"
      />
    </form>
  </section>
</DocumentFragment>
`;
